File: tools/improve/tests/original/maudec/py/nat-funs.py


def fib(a1: int) -> int:
	match a1:
		case 0:
			return 0

		case 1:
			return 1

		case N if N >= 2:
			return fib(N - 2) + fib((N - 2) + 1)



def _u0021(a1: int) -> int:
	match a1:
		case 0:
			return 1

		case N if N >= 1:
			return ((N - 1) + 1) * _u0021(N - 1)



def u0024gcd(M: int, N: int) -> int:
	match N:
		case 0:
			return M

		case N:
			return u0024gcd(N, M % N)

File: tools/improve/tests/original/maudec/cpp/nat-funs.cpp



int fib(int a1) {
	if (a1 == 0) {
		return 0;
	}
	if (a1 == 1) {
		return 1;
	}
	if (a1 >= 2) {
		return fib(a1 - 2) + fib((a1 - 2) + 1);
	}
}

int _u0021(int a1) {
	if (a1 == 0) {
		return 1;
	}
	if (a1 >= 1) {
		return ((a1 - 1) + 1) * _u0021(a1 - 1);
	}
}
